backend/app/mcp_server/tool_registry.py

```
@@ -0,0 +1,342 @@
+    Args:
+        func: Async function to call
+        kwargs: Keyword arguments for the function
+
+    Returns:
+        Result from the async function
+    """
+    try:
+        # Check if there's already a running event loop
+        loop = asyncio.get_running_loop()
+        logger.debug(
+            f"[MCP] Event loop already running, using thread-based execution"
```

Ruff (0.14.14)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)